platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/model/internal/deterministic/DeterministicHeartbeatScheduler.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.common.wiring.model.internal.deterministic;
+
+import static com.swirlds.common.utility.CompareTo.isGreaterThanOrEqualTo;
+
+import com.swirlds.base.time.Time;
+import com.swirlds.common.wiring.model.TraceableWiringModel;
+import com.swirlds.common.wiring.model.internal.standard.AbstractHeartbeatScheduler;
+import com.swirlds.common.wiring.model.internal.standard.HeartbeatTask;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+/**
+ * A deterministic implementation of a heartbeat scheduler.
+ */
+public class DeterministicHeartbeatScheduler extends AbstractHeartbeatScheduler {
+
+    /**
+     * This maps from the period of the heartbeat to the list of heartbeats that want to be notified with that period.
+     */
+    private final Map<Duration, List<HeartbeatTask>> heartbeatsByPeriod = new HashMap<>();
+
+    /**
+     * The timestamps of the last heartbeats sent for each group of heartbeats that subscribe to the same period.
+     */
+    private final Map<Duration, Instant> lastHeartbeats = new HashMap<>();
```

Comment:
Nit: I think the term "previous" might be preferable, since "last" has an alternate meaning

Reply:
renamed as suggested